### PR TITLE
chore(stdlib): Add docs to the Pervasives module

### PR DESCRIPTION
**Pull request by phated**
This relies on a bunch of the other PRs that I submitted, so marking it as draft until I can rebase.

This updates the Pervasives module with graindoc so we don't wipe away the docs when we auto-generate them for the website.

### CODE REVIEW

**Review by marcusroberts — APPROVED** [review]
Approving this draft, the changes look good!

**Comment by phated** [comment]
Updated the base, as only #1301 is required before this is promoted from draft.

**Comment by phated** [comment]
Ready for review!

File: stdlib/pervasives.gr [hunk 1]
```diff
@@ -58,71 +414,158 @@ export (>>)
...
+ *
+ * @since v0.4.0
  */
 export let cons = (a, b) =>
   [
     a,
     ...b
   ] // <- workaround for (grain-lang/grain#802) [TODO] fix #802 and delete
+/**
+ * The empty list syntax (`[]`) provided as a value;
```
Comment (ospencer) [inline stdlib/pervasives.gr:561]:
```suggestion
 * The empty list syntax (`[]`) provided as a value.
```

File: stdlib/pervasives.gr [hunk 2]
```diff
@@ -58,71 +414,158 @@ export (>>)
...
+ */
 
-export enum List<a> {
-  [],
-  [...](a, List<a>),
-}
+/**
+ * Creates a box with the same type as the given initial value.
+ * Values inside a box can be swapped out with the `:=` operator.
+ * Generally, `let mut` expressions will be preferable to using a Box.
```
Comment (ospencer) [inline stdlib/pervasives.gr:521]:
```suggestion
 * Creates a box containing the given initial value.
 * Values inside a box can be swapped out with the `:=` operator.
 * Generally, `let mut` expressions are preferable to using a Box.
```

File: stdlib/pervasives.gr [hunk 3]
```diff
@@ -58,71 +414,158 @@ export (>>)
...
+ * @since v0.3.0
+ */
 export primitive throw: Exception -> a = "@throw"
+
+/**
+ * Unconditionally throw a `Failure` exception with a message.
+ * Currently, Exceptions cannot be caught and will crash your program.
+ *
+ * @param message: The reason for the failure
+ * @returns Anything and nothing—your program won't continue past a throw
```
Comment (ospencer) [inline stdlib/pervasives.gr:496]:
```suggestion
 * @returns Anything and nothing—your program won't continue past a fail expression
```

File: stdlib/pervasives.gr [hunk 4]
```diff
@@ -58,71 +414,158 @@ export (>>)
...
 
 // Exceptions
 export exception Failure(String)
 export exception InvalidArgument(String)
 
-// Other operations
-export primitive ignore: a -> Void = "@ignore"
-export primitive assert: Bool -> Void = "@assert"
+/**
+ * Throw an exception. Currently, Exceptions cannot be caught and will crash your program.
```
Comment (ospencer) [inline stdlib/pervasives.gr:482]:
```suggestion
 * Throw an exception. Currently, exceptions cannot be caught and will crash your program.
```

**Review by ospencer — APPROVED** [review]

**Pull request merged**